algorithms/strings.py

```diff
@@ -52,3 +52,19 @@ def _get_key(s: str):
             grouped[key].append(x)
 
         return [x for x in grouped.values()]
+
+    # https://leetcode.com/problems/simplify-path/
+    def simplifyPath(self, path: str):
+        res = []
+
+        spl = path.split('/')
+        for x in spl:
+            if not x or x == '.':
+                continue
+            if x == '..':
+                if len(res):
+                    del res[len(res) - 1]
```

Comment:
yep. just a suggestion: you can also use `res.pop()` to remove last element, or `del res[-1]`.